lib/LaTeXML/Package/expl3.lua.ltxml

```diff
@@ -0,0 +1,124 @@
+      my $n       = $gullet->readNumber;
+      my $address = '__intarray' . '_' . ToString($tok) . '_' . $n->valueOf;
+      my $table   = LookupValue($address);
+      if (!$table) {                            # Create new table, if none
+        $table = [];
+        AssignValue($address => $table, 'global'); }
+      return $table; }
+    else {
+      $gullet->unread($tok);
+      Error('expected', '__intarray:w', $gullet,
+        "Expected an intarray identifier, got " . ToString($tok));
+      return []; } });
+
+DefPrimitiveI('\__intarray:w', 'Number', sub {
+    Error('unexpected', '\__intarray:w', $_[0], "Unexpected isolated \\__intarray:w?");
+    return; },
+  protected => 1);
+
+DefPrimitiveI('\__intarray_gset_count:Nw', 'Intarray Number', sub {
+    my ($stomach, $table, $newlength) = @_;
+    my $length = scalar(@$table);
+    $newlength = $newlength->valueOf;
+    $$table[$newlength - 1] = 0;                     # Grow in one step for efficiency
+    for (my $i = $length + 1 ; $i <= $newlength ; $i++) {
+      $$table[$i - 1] = 0; }
+    return; },
+  protected => 1);
+
+DefMacroI('\intarray_count:N', 'Intarray', sub {
+    my ($gullet, $table) = @_;
+    return Explode(scalar(@$table)); });
+
+DefMacroI('\__intarray_gset:wF', 'Number Intarray Number {}', sub {
+    my ($stomach, $pos, $table, $value, $ifmissing) = @_;
+    my $length = scalar(@$table);
+    $pos = $pos->valueOf;
+    if (($pos > 0) && ($pos < $length)) {
+      $$table[$pos - 1] = $value->valueOf; }
+    else {
+      return $ifmissing; }
+    return; },
+  protected => 1);
+
+DefMacroI('\__intarray_gset:w', 'Number Intarray Number', sub {
+    my ($stomach, $pos, $table, $value) = @_;
+    my $length = scalar(@$table);
+    $pos = $pos->valueOf;
+    if ($pos > 0) {
+      $$table[$pos - 1] = $value->valueOf; }
+    return; },
+  protected => 1);
+
+DefPrimitiveI('\intarray_gzero:w', 'Intarray', sub {
+    my ($stomach, $table) = @_;
+    my $length = scalar(@$table);
+    for (my $i = 1 ; $i <= $length ; $i++) {
+      $$table[$i - 1] = 0; }
+    return; },
+  protected => 1);
+
+DefMacroI('\__intarray_item:wF', 'Number Intarray {}', sub {
+    my ($stomach, $pos, $table, $ifmissing) = @_;
+    my $length = scalar(@$table);
+    $pos = $pos->valueOf;
+    if (($pos > 0) && ($pos < $length)) {
+      return Explode($$table[$pos - 1]); }
+    else {
+      return $ifmissing; } });
+
+DefMacroI('\__intarray_item:w', 'Number Intarray', sub {
+    my ($stomach, $pos, $table) = @_;
+    my $length = scalar(@$table);
+    $pos = $pos->valueOf;
+    if (($pos > 0) && ($pos < $length)) {
+      return Explode($$table[$pos - 1]); }
+    return; });
+
+DefMacroI('\__intarray_to_clist:Nn', 'Intarray', sub {
+    my ($stomach, $table) = @_;
+    return Tokenize(join(', ', @$table)); });
+
+DefMacroI('\__intarray_range_to_clist:w', 'Intarray Number Number', sub {
+    my ($stomach, $table, $from, $to) = @_;
+    $from = $from->valueOf - 1;
+    $to   = $to->valueOf - 1;
```

Comment:
This is a rare edge case, but if either of the input numbers happens to be (erroneously?) zero `0`, the `$from` or `$to` index will be `-1`, which will lead to a rather peculiar indexing in perl. It may be worth guarding them to be within the allowable range of `$table` here as well.

Reply:
It seems like the expl3 code usually has the public functions do the input validation, and then the `__private` functions don't need to worry about that.  The corresponding lua code doesn't validate the inputs.  (I've also not noticed a public version of this function, so it might all be moot.)

Reply:
ditto.